File: services/core-api/internal/features/platformstatus/service.go
// Service layer: business shaping, deadlines, no HTTP, no SQL. Version-NEUTRAL —
// /v1 and /v2 handlers share this service unchanged (research A3).
package platformstatus

import (
	"context"
	"time"
)

const readTimeout = 3 * time.Second

// PlatformStatus is the domain model (data-model.md §4).
type PlatformStatus struct {
	Environment       string
	DatabaseName      string
	DatabaseTime      time.Time
	MigrationVersion  int64
	MigrationsApplied int64
}

// StatusReader is the repository seam; hand-rolled fakes implement it in tests.
type StatusReader interface {
	Status(ctx context.Context) (PlatformStatus, error)
}

type Service struct {
	repo StatusReader
	env  string
}

func NewService(repo StatusReader, env string) *Service {
	return &Service{repo: repo, env: env}
}

func (s *Service) Get(ctx context.Context) (PlatformStatus, error) {
	ctx, cancel := context.WithTimeout(ctx, readTimeout)
	defer cancel()

	status, err := s.repo.Status(ctx)
	if err != nil {
		return PlatformStatus{}, err
	}
	status.Environment = s.env
	return status, nil
}

File: apis/edge-api/store/src/status/service.ts
// Service layer: domain logic and shaping — no HTTP, no SQL. Version-NEUTRAL: the v1
// and v2 handlers consume this service unchanged (research A3).
import type { StatusRepository } from "./repository";
import type { PlatformStatus } from "./types";

export interface PlatformStatusService {
  getStatus(): Promise<PlatformStatus>;
}

export function createPlatformStatusService(
  repo: StatusRepository,
  environment: string,
): PlatformStatusService {
  return {
    async getStatus(): Promise<PlatformStatus> {
      const status = await repo.status();
      return { ...status, environment };
    },
  };
}
